### DISCUSSION TITLE
Unable to Install on Kali Linux Rolling

### DISCUSSION THREAD

**Discussion by pauljackman2023** [ainfosec/FISSURE#47] [Hard Questions]
Hi All

I am trying to install Fissure on Kali Linux rolling running under WSL2.  I have cloned the repository etc and following the install instructions i get the below

┌──(root2㉿DESKTOP-65G8UE6)-[~/FISSURE]
└─$ sudo git checkout <Python2_maint-3.7> or <Python3_maint-3.8> or <Python3_maint-3.10>
-bash: syntax error near unexpected token `newline'

┌──(root2㉿DESKTOP-65G8UE6)-[~/FISSURE]
└─$

Am I doing something wrong or am I being thick, I am fairly new to using linux environments, I do not wish to change distros either, as I very much like KEX-WIN under Kali allowing me to run the full xfce gui in a window inside windows.

![image](https://github.com/ainfosec/FISSURE/assets/141925107/2ce5c9a6-ebb9-47a1-a3c1-aa5f0f7cd3a7)
![image](https://github.com/ainfosec/FISSURE/assets/141925107/a25562b7-dce4-41c8-a68e-4365d249d568)

If it is just a case of needing to install a different version of python, if someone could post the repository i need ideally a git repository so i can clone and install that'd be great. Step by step instructions on any suggested fixes would be very much appreciated as I said I am fairly new to using Linux terminals etc,  my only other experience of these sorts of systems was back in the bad old days of UNIX on IBM mainframes and I always had idiot proof lists of commands I had to run lol.

**Comment by cpoore1** [comment]
Try checking out just one of those options in the command. Kali uses the Python3_maint-3.10 branch. ‘git checkout Python3_maint-3.10’ and I recommend not using sudo during the install process.

**Reply by pauljackman2023** [reply]
┌──(root2㉿DESKTOP-65G8UE6)-[~]
└─$ git checkout Python3_maint-3.10
fatal: not a git repository (or any of the parent directories): .git

┌──(root2㉿DESKTOP-65G8UE6)-[~]
└─$ git checkout <Python3_maint-3.10>
-bash: syntax error near unexpected token `newline'

get these errors, not sure what i'm doing wrong, usually linux is much more verbose, i can usually guess what i'm doing wrong or sometimes, it tells me exactly what i'm doing wrong

**Reply by pauljackman2023** [reply]
Following a suggestion in a debian forum, I am rebuilding my python from the 3.10 source package to make sure it's installed and created correctly

**Reply by cpoore1** [reply]
This attempt didn’t work because the current directory was not ~/FISSURE. That directory contains the hidden .git folder. The <> characters should not be entered. When cloning a git repository you can check out the branches by typing the exact characters listed in the drop-down menu on the GitHub page. If no changes were made locally before the checkout command is issued, it should replace all local files from the default branch with files from the specified branch.

**Comment by pauljackman2023** [comment]
Hi, I actually managed to do it by watching the installation instructions video, one of the commands i was copying was formatted slightly wrong, when I type it all manually I was able to switch the Python Version to the correct one.  Anyone having similar issues should watch https://www.youtube.com/watch?v=PGIZHhLswXg&list=PLs4a-ctXntfjpmc_hrvI0ngj4ZOe_5xm_
